urllib3/exceptions.py

```diff
@@ -196,6 +196,14 @@ class ResponseNotChunked(ProtocolError, ValueError):
     pass
 
 
+class BodyNotHttplibCompatible(AssertionError, ValueError):
```

Comment:
Sorry, why does HTTPWarning need to inherit from Urllib3Exception? I think this change was overbroad. HTTPError is already the base exception class for urllib3, so the only change required was to not inherit from AssertionError or ValueError, and just to inherit from HTTPError.